lib/create-testing-library-rule/detect-testing-library-utils.ts

```diff
@@ -595,6 +622,41 @@ export function detectTestingLibraryUtils<
         }
       );
 
+    const isCreateEventUtil: IsCreateEventUtil = (node) => {
+      const isCreateEventCallback = (
+        identifierNodeName: string,
+        originalNodeName?: string
+      ) => [identifierNodeName, originalNodeName].includes(CREATE_EVENT_NAME);
+      if (ASTUtils.isIdentifier(node)) {
+        return isPotentialTestingLibraryFunction(node, isCreateEventCallback);
+      }
+      if (ASTUtils.isIdentifier(node.callee)) {
+        return isPotentialTestingLibraryFunction(
+          node.callee,
+          isCreateEventCallback
+        );
+      }
+      if (!isMemberExpression(node.callee)) {
+        return false;
+      }
+      if (ASTUtils.isIdentifier(node.callee.object)) {
+        return isPotentialTestingLibraryFunction(
+          node.callee.object,
+          isCreateEventCallback
+        );
+      }
+      if (
```

Comment:
You don't need to manually check the type of the node and navigate its properties to get the deepest identifier! You can do that directly with `getDeepestIdentifierNode`.

Could you check how it's used in other parts of the code and simplify this function with it?

Reply:
cool! I'll check it out. This function was modified a couple of times (I built it iteratively) so while I knew about that function I think that eventually the final form of this function was like this and I totally forgot about that function 😅

Reply:
@Belco90  I applied the suggestion; it doesn't solve all the cases but I was able to simplify two of those. Not in all cases  I want to get the deepest identifier node; see examples below

```ts
// here getDeepestIdentifierNode would retrieve the "click" node, but I need the "createEvent" node
const event = createEvent.click()

// Same here, I need "createEvent" but I'd get "click"
const event = dom.createEvent.click()

// here it works as expected
const event = dom.createEvent('click')

// here the same
const event = createEvent('click')
```

Applied here https://github.com/testing-library/eslint-plugin-testing-library/pull/398/commits/57622f0576989c7bd76a507269e196b24a2ae8be

Reply:
Cool, it makes sense. At least it simplified traversing the node a bit.